### PR TITLE
feat(test): setup vitest and add unit tests for formatter utilities (closes #3)

### LINKED ISSUE #3: Setup testing framework and add initial unit tests
### Description
The codebase currently lacks a testing framework and unit tests. Since this is a Vite project, we should install and configure a testing framework like `vitest`.

### Objective
1. Install and configure `vitest` (or a similar suitable framework).
2. Write a few simple unit tests for the standalone utility functions to help prevent future regressions.

### Files to check
- `package.json`
- `src/lib/utils/formatters.ts` (create an adjacent `formatters.test.ts` file)

### PR DESCRIPTION
**Pull request by anasahhm**
Closes #3 by setting up Vitest and adding initial unit tests for formatter utilities.

## Includes
- Vitest configuration
- Test script in package.json
- Unit tests covering formatting and edge cases

### CODE REVIEW

File: src/lib/utils/formatters.test.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  complexityLabel,
+  friendlinessLabel,
+  progressLabel,
+  statusLabel,
+} from './formatters';
+
+describe('formatters utilities', () => {
+  it('formats ISO date correctly', () => {
+    expect(formatTimestamp('2024-01-15T00:00:00Z')).toBe('Jan 15, 2024');
+  });
```
[the remark this answers is not in the document]
Comment (anasahhm) [reply src/lib/utils/formatters.test.ts:25]: Good catch ! 
Updated tests to use fixed timestamps and Vitest fake timers to ensure deterministic and timezone-safe behavior.

**Comment by vee1e** [comment]
Thanks, please show some screenshots or terminal output demonstrating the tests passing and we can review further.

**Comment by anasahhm** [comment]
tests

Here’s the test run output demonstrating all tests passing locally:

- 13/13 tests passed successfully  
- Includes deterministic time-based tests using Vitest fake timers  
- Verified using `vitest --run` for a clean, non-watch execution

**Comment by vee1e** [comment]
thanks

**Comment by vee1e** [comment]
vouch @anasahhm

**Comment by anasahhm** [comment]
Thanks for the vouch, Glad I could contribute!
